pkg/monitor/dynkeepalived.go

```diff
@@ -63,6 +63,32 @@ func KeepalivedWatch(kubeconfigPath, clusterConfigPath, templatePath, cfgPath st
 				}
 			}
 			prevConfig = &newConfig
+
+			// Signal to keepalived whether the haproxy firewall rule is in place
+			ruleExists, err := checkHAProxyPreRoutingRule(apiVip.String(), apiPort, lbPort)
+			if err != nil {
+				log.Error("Failed to check for haproxy firewall rule")
+			} else {
+				filePath := "/var/run/keepalived/iptables-rule-exists"
```

Comment:
I guess it will be better to set filename only once and not every loop iteration

Reply:
IIRC, Keepalived monitor runs also in worker nodes, I guess we shouldn't run this check for worker nodes.

Maybe I'm missing something here, but it seems to me that haproxy-monior is the right place to add this logic.

Reply:
haproxy-monitor doesn't share its /var/run mount with keepalived. I would also argue that haproxy doesn't care about this file. It's consumed by keepalived, so it should be done in the keepalived monitor.

Reply:
Oh, and I moved the filename to a const at the top of the file. Good suggestion.